packages/web/src/js/Offcanvas.ts

```diff
@@ -0,0 +1,134 @@
+import BaseComponent from './BaseComponent';
+import EventHandler from './dom/EventHandler';
+import { enableToggleTrigger, enableDismissTrigger } from './utils/ComponentFunctions';
+
+const NAME = 'offcanvas';
+const DATA_KEY = 'offcanvas';
+const EVENT_KEY = `.${DATA_KEY}`;
+
+const EVENT_SHOW = `show${EVENT_KEY}`;
+const EVENT_SHOWN = `shown${EVENT_KEY}`;
+const EVENT_HIDE = `hide${EVENT_KEY}`;
+const EVENT_HIDDEN = `hidden${EVENT_KEY}`;
+
+const OFFCANVAS_BREAKPOINT = 1280;
```

Comment:
Ideally, we should take these from the design tokens. Options I can think of now:

1. Manually manage the smallest possible set of design tokens for JS inside `/packages/design-tokens/src/js`.
2. Store these values in constants somewhere in `/packages/web/src/js`.

The first seems better to me.

I think we are only talking about screen breakpoints at this moment so both ways are OK for me. I don't think we need any automation like Styled Dictionary at the time being.